src/plugins/intel_gpu/src/graph/include/primitive_inst.h

```diff
@@ -541,6 +541,15 @@ class typed_primitive_inst_base : public primitive_inst {
             typ_node.get_users().front()->can_be_optimized()) {  // check if the only user is concat
             return false;
         }
+
+        auto& e = typ_node.get_program().get_engine();
+        auto max_mem = e.get_max_memory_size();
```

Comment:
Wouldn't checking of max **used** memory be better here?

Reply:
I think we can do like this 
- fix ocl_engine::check_allocatable not to throw exception but return false if not allocatable 
- add the required assertion in ocl_engine::allocate_memory() like OPENVINO_ASSERT(check_allocatable(layout,type), true)
- use check_allocatable here

Reply:
Though basically I think chcking used memory here for dynamic shape is not that much meaningful because the filtering here is to avoid too much memory for bounded shape while actual exception will be happen eventually in the allocate_output function.

Reply:
Updated! Thanks!